### PR TITLE
Customizer UI Improvment

### PR DESCRIPTION
**Pull request by MichaelPFrey**
* only enable the customizer UI when ready 
* descriptionLoD "Description Only" (hiding the variable name when a description is set) 
* set a flag when we have an unreadable json file 
* ask before overwriting unreadable JSON-Files

see also #2316

### CODE REVIEW

File: src/parameter/ParameterWidget.cc [hunk 1]
```diff
@@ -388,6 +396,24 @@ void ParameterWidget::updateParameterSet(std::string setName)
...
 	}
 }
+
+void ParameterWidget::writeParameterSets()
+{
+	if(this->unreadableFileExists){
+		bool ok = true;
+		QMessageBox msgBox;
+		msgBox.setText(_("Saving presets"));
+		msgBox.setInformativeText(QString(_("%1 was found, but was unreadble. Do you want to overwrite %1?")).arg(QString::fromStdString(jsonFile)));
```
Comment (kintel) [inline src/parameter/ParameterWidget.cc]: `this->jsonFile`

File: src/parameter/ParameterWidget.cc [hunk 2]
```diff
@@ -388,6 +396,24 @@ void ParameterWidget::updateParameterSet(std::string setName)
...
 		}
-		writeParameterSet(this->jsonFile);
+		writeParameterSets();
 	}
 }
+
+void ParameterWidget::writeParameterSets()
+{
+	if(this->unreadableFileExists){
+		bool ok = true;
```
Comment (kintel) [inline src/parameter/ParameterWidget.cc]: Looks like `ok` is never used

File: src/parameter/ParameterWidget.cc [hunk 3]
```diff
@@ -388,6 +396,24 @@ void ParameterWidget::updateParameterSet(std::string setName)
...
+		QMessageBox msgBox;
+		msgBox.setText(_("Saving presets"));
+		msgBox.setInformativeText(QString(_("%1 was found, but was unreadble. Do you want to overwrite %1?")).arg(QString::fromStdString(jsonFile)));
+		msgBox.setStandardButtons(QMessageBox::Save | QMessageBox::Cancel);
+		msgBox.setDefaultButton(QMessageBox::Cancel);
+		int ret = msgBox.exec();
+
+		if (ret == QMessageBox::Cancel) {
+			return;
+		}
```
Comment (kintel) [inline src/parameter/ParameterWidget.cc]: You could probably make the above a tiny bit cleaner by doing:
`if (msgBox.exec() == QMessageBox::Cancel) return;`

File: src/parameter/parameterslider.cpp [hunk 4]
```diff
@@ -13,14 +13,21 @@ ParameterSlider::ParameterSlider(ParameterObject *parameterobject, int showDescr
 	connect(slider, SIGNAL(sliderReleased()), this, SLOT(onReleased()));
 	connect(slider, SIGNAL(valueChanged(int)), this, SLOT(onSliderChanged(int)));
 	connect(doubleSpinBox, SIGNAL(valueChanged(double)), this, SLOT(onSpinBoxChanged(double)));
-	if (showDescription == 0) {
+	if (showDescription == 0 || showDescription == 3) {
```
Comment (kintel) [inline src/parameter/parameterslider.cpp]: This may be a good refactoring for later: Use an enum instead of the magic `showDescription` integer

Comment (MichaelPFrey) [reply src/parameter/parameterslider.cpp]: entered into #2316

**Comment by MichaelPFrey** [comment]
thank you for the review - changed as suggested.

**Pull request merged**